connector/connect/server/src/main/scala/org/apache/spark/sql/connect/ml/AlgorithmRegisty.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connect.ml
+
+import org.apache.spark.connect.proto
+import org.apache.spark.ml
+import org.apache.spark.ml.{Estimator, Model}
+import org.apache.spark.ml.classification.TrainingSummary
+import org.apache.spark.sql.DataFrame
+
+
+object AlgorithmRegistry {
+
+  def get(name: String): Algorithm = {
+    name match {
+      case "LogisticRegression" => new LogisticRegressionAlgorithm
+      case _ =>
+        throw new IllegalArgumentException()
+    }
+  }
+
+}
+
+
+abstract class Algorithm {
+
+  def initiateEstimator(uid: String): Estimator[_]
+
+  def getModelAttr(model: Model[_], name: String): Either[proto.MlCommandResponse, DataFrame]
```

Comment:
why not making `DataFrame`/`Relation` an `oneof` item in `MlCommandResponse` ?

Reply:
Finally, I think I understand it.

`ModelAttr` may return a `MlCommandResponse` or `DataFrame`.

`ModelAttr` is used in `MlCommand` to get a scalar result, or `MlRelation` to get a logical plan.

maybe here we can split `getModelAttr` (/ `getModelSummaryAttr `) into two methods `getModelScalarAttr` and `getModelDataFrameAttr` to make it a bit easier to understand?

Reply:
> maybe here we can split getModelAttr (/ getModelSummaryAttr ) into two methods getModelScalarAttr and getModelDataFrameAttr to make it a bit easier to understand?

This is what I did in initial code, but then I changed it to current code. Because:

in ML code part, returning either MlCommandResponse or DataFrame either , the logic for them is the same,
if we split them into 2 methods, the model summary, we have complex hierarchy, then each super class we need to define 2 methods similarly, it makes code bloating and hard to maintain.

Reply:
ok. I am neutral on it.